cmd/influx_inspect/export.go

```diff
@@ -1,92 +1,143 @@
-		fmt.Println("Opening", f, "in database", database)
-		temp, _ := ioutil.TempDir("/tmp", "influxdb")
+func (c *cmdExport) run() error {
+	if err := c.validate(); err != nil {
+		return err
+	}
 
-		if err != nil {
-			fmt.Printf("%v", err)
-			os.Exit(1)
-		}
+	return c.export()
+}
 
-		reader, err := tsm1.NewTSMReader(file)
+func (c *cmdExport) export() error {
+	if err := c.walkFiles(); err != nil {
+		return err
+	}
+	return c.writeFiles()
+}
 
+func (c *cmdExport) walkFiles() error {
+	err := filepath.Walk(c.path, func(path string, f os.FileInfo, err error) error {
 		if err != nil {
-			fmt.Printf("%v", err)
-			os.Exit(1)
+			return err
 		}
-
-		filename := temp + "/" + database + ".lp"
-
-		of, _ := os.Create(filename)
-		defer of.Close()
-		w := bufio.NewWriter(of)
-
-		fmt.Println("Writing", reader.KeyCount(), "series to", filename)
-
-		for i := 0; i < reader.KeyCount(); i++ {
-			var pairs string
-			key, typ := reader.KeyAt(i)
-			values, _ := reader.ReadAll(key)
-			split := strings.Split(key, "#!~#")
-			measurement, field := split[0], split[1]
-
-			for _, value := range values {
-				switch typ {
-				case tsm1.BlockFloat64:
-					pairs = field + "=" + fmt.Sprintf("%v", value.Value())
-				case tsm1.BlockInteger:
-					pairs = field + "=" + fmt.Sprintf("%vi", value.Value())
-				case tsm1.BlockBoolean:
-					pairs = field + "=" + fmt.Sprintf("%v", value.Value())
-				case tsm1.BlockString:
-					pairs = field + "=" + fmt.Sprintf("\"%v\"", value.Value())
-				default:
-					pairs = field + "=" + fmt.Sprintf("%v", value.Value())
+		if filepath.Ext(path) == c.ext {
+			//files = append(files, path)
+			relPath, _ := filepath.Rel(c.path, path)
+			dirs := strings.Split(relPath, string(byte(os.PathSeparator)))
+			if dirs[0] == c.database || c.database == "" {
+				if dirs[1] == c.retentionPolicy || c.retentionPolicy == "" {
+					key := filepath.Join(dirs[0], dirs[1])
+					files := c.files[key]
+					if files == nil {
+						files = []string{}
+					}
+					c.files[key] = append(files, path)
 				}
-
-				fmt.Fprintln(w, measurement, pairs, value.UnixNano())
 			}
+		}
+		return nil
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+	return nil
+}
 
-			w.Flush()
+func (c *cmdExport) writeFiles() error {
+	// open our output file and create an output buffer
+	var w io.WriteCloser
+	w, _ = os.Create(c.out)
+	defer w.Close()
+	if c.compress {
+		w = gzip.NewWriter(w)
+	}
+	fmt.Fprintln(w, "# DML")
+
+	for key, files := range c.files {
+		keys := strings.Split(key, string(byte(os.PathSeparator)))
+		fmt.Fprintf(w, "# CONTEXT-DATABASE:%s\n", keys[0])
+		fmt.Fprintf(w, "# CONTEXT-RETENTION-POLICY:%s\n", keys[1])
+		for _, f := range files {
+			file, err := os.OpenFile(f, os.O_RDONLY, 0600)
+			if err != nil {
+				return fmt.Errorf("%v", err)
+			}
+			defer file.Close()
+			reader, err := tsm1.NewTSMReader(file)
+			if err != nil {
+				return err
+			}
+			defer reader.Close()
+
+			for i := 0; i < reader.KeyCount(); i++ {
```

Comment:
This is the best way.  It arrives as a `Point`, but it's not stored or retrieved in that format ever again.